### PR TITLE
cleanup: simplify `rbdGetDeviceList()`

### PR DESCRIPTION
**Pull request by nixpanic**
The `rbdGetDeviceList()` function uses two very similar types for
converting krbd and NBD device information from JSON. There is no need
to use this distinction, and callers of `rbdGetDeviceList()` should not
need to care about it either.

By introducing a `deviceInfo` interface with Get-functions, the
`rbdGetDeviceList()` function becomes a little simpler, with a clearly
defined API for the returned list.

### CODE REVIEW

File: internal/rbd/rbd_attach.go [hunk 1]
```diff
@@ -91,25 +90,43 @@ var (
...
 
+type deviceInfo interface {
+	GetName() string
+	GetPool() string
+	GetRadosNamespace() string
+	GetDevice() string
+}
+
 // rbdDeviceInfo strongly typed JSON spec for rbd device list output (of type krbd).
 type rbdDeviceInfo struct {
```
Comment (Rakshith-R) [inline internal/rbd/rbd_attach.go:101]: We no longer have to save ID ? 
It's not used anywhere ?

Comment (nixpanic) [reply internal/rbd/rbd_attach.go:101]: Indeed, it is not used anywhere for attaching/mounting.

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@Mergifyio queue

**Pull request merged**